Tensile/SolutionStructs.py

```diff
@@ -2420,9 +2420,16 @@ def isDirectToLdsDoable(state, tc):
     #TN
     # use for all precisions with TransposeLDS=1
 
-    if state["ProblemType"]["DataType"].isHalf() and state["AssertSummationElementMultiple"] % (2 * state["GlobalLoadVectorWidth%c"%tc])  != 0:
-      reject(state, "can't use DirectToLds for FP16 with AssertSummationElementMultiple %u" % state["AssertSummationElementMultiple"])
-      return False
+    if state["ProblemType"]["DataType"].isHalf():
+      if state["AssertSummationElementMultiple"] % (2 * state["GlobalLoadVectorWidth%c"%tc])  != 0:
+        reject(state, "can't use DirectToLds for FP16 with AssertSummationElementMultiple %u" % state["AssertSummationElementMultiple"])
+        return False
+      if state["ProblemType"]["TransposeA"] != True or state["ProblemType"]["TransposeB"] != False:
```

Comment:
Probably, we need to improve this part.
Actually, I changed isDirectToLdsDoable to reject instead of just changing DirectToLds = 0.
This is because I did not want to run same combinations by converting DTL to 0.
However, I realized that this prevented enabling DirectToLds for only A or B.
In this case, we should be able to apply DirectToLds for A if TransposeA=false (regardless of B).

I think it is better to change back this part not to reject code here.

Reply:
For hgemm, DirectToLds only works for TN, all other transpose combos currently fail validation. So the check for specific transpose types is accurate in this case.

Reply:
OK. We cannot combine like DirectToLdsA= True with DirectToLds=B.
For HGEMM only?
For other types, I think we should be able to do that if we changed the reject logic in this function.

Reply:
I understand we do not support other types for now, but technically we can support them in the future.